ingress/controllers/nginx/README.md

```diff
@@ -186,60 +187,363 @@ data:
 Please check the [udp services](examples/udp/README.md) example
 
 
-## Custom NGINX configuration
+### Annotations
 
-Using a ConfigMap it is possible to customize the defaults in nginx.
+|Name                 |type|
+|---------------------------|------|
+|[ingress.kubernetes.io/rewrite-target](#rewrite)|URI|
+|[ingress.kubernetes.io/add-base-url](#rewrite)|true or false|
+|[ingress.kubernetes.io/limit-connections](#rate-limiting)|number|
+|[ingress.kubernetes.io/limit-rps](#rate-limiting)|number|
+|[ingress.kubernetes.io/auth-type](#authentication)|basic or digest|
+|[ingress.kubernetes.io/auth-secret](#authentication)|string|
+|[ingress.kubernetes.io/auth-realm](#authentication)|string|
+|[ingress.kubernetes.io/ssl-redirect](#server-side-https-enforcement-through-redirect)|true or false|
+|[ingress.kubernetes.io/upstream-max-fails](#custom-nginx-upstream-checks)|number|
+|[ingress.kubernetes.io/upstream-fail-timeout](#custom-nginx-upstream-checks)|number|
+|[ingress.kubernetes.io/secure-upstream](#secure-nginx-upstreams)|true or false|
+|[ingress.kubernetes.io/whitelist-source-range](#whitelist-source-range)|CIDR|
 
-Please check the [tcp services](examples/custom-configuration/README.md) example
+### Custom NGINX upstream checks
 
+NGINX exposes some flags in the [upstream configuration](http://nginx.org/en/docs/http/ngx_http_upstream_module.html#upstream) that enables the configuration of each server in the upstream. The ingress controller allows custom `max_fails` and `fail_timeout` parameters in a global context using `upstream-max-fails` or `upstream-fail-timeout` in the NGINX Configmap or in a particular Ingress rule. By default this values are 0. This means NGINX will respect the `readinessProbe`, if is defined. If there is no probe, NGINX will not mark a server inside an upstream down.
 
-## Custom NGINX template
+**With the default values NGINX will not health check your backends, and whenever the endpoints controller notices a readiness probe failure that pod's ip will be removed from the list of endpoints, causing nginx to also remove it from the upstreams.**
```

Comment:
I'd move this up into the health checks section, and follow it up with: if you want to customize your health checks, specify `upstream-max-fails` and `upstream-fail-timeout`, as shown in this [example](link).